lunabot_control/src/mpc.cpp

```diff
@@ -159,59 +105,75 @@ double MPC::smooth_clamp_(double x, double a, double b) {
   return a + (b - a) * (1 + sin(M_PI * x / 2)) / 2;
 }
 
-std::vector<Eigen::MatrixXd> MPC::normal_distribute_(Eigen::MatrixXd means,
-                                                     Eigen::MatrixXd std_devs, int count) {
-  std::vector<Eigen::MatrixXd> random_vels(count);
+std::pair<std::vector<Eigen::MatrixXd>, std::vector<Eigen::MatrixXd>>
+MPC::normal_distribute_(Eigen::MatrixXd means, Eigen::MatrixXd std_devs, int count) {
+  std::vector<Eigen::MatrixXd> random_vels_raw(count);
+  std::vector<Eigen::MatrixXd> random_vels_clamped(count);
+
+#pragma omp parallel for
```

Comment:
need to officially benchmark. i couldn't tell. it seems to still run at 5-7hz. also our benchmarks should be done on the jetson because that's a better reflection of the multiprocessing that can be done. i also personally think we should do gpu acceleration rather than cpu acceleration.